crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -58,12 +56,18 @@ use fuel_gas_price_algorithm::{
     },
 };
 use futures::FutureExt;
+use std::{
+    num::NonZeroU64,
+    sync::Arc,
+};
 use tokio::sync::broadcast::Receiver;
 
 /// The service that updates the gas price algorithm.
 pub struct GasPriceServiceV1<L2, DA, StorageTxProvider> {
     /// The algorithm that can be used in the next block
     shared_algo: SharedV1Algorithm,
+    /// The latest gas price
+    latest_gas_price: Arc<parking_lot::RwLock<(u32, u64)>>,
```

Comment:
I wonder if it's best to move the `ArcGasPriceEstimate` type to `fuel-types` or some other places that is accessible from the gas_price_service? 
The implementation of the trait `GasPriceEstimate` can remain in `fuel_core`

Reply:
edit: I guess I missed the point of `ArcGasPriceEstimate`, closing